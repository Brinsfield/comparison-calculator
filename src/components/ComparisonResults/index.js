import React, { Component } from 'react';
import logo from '../../assets/logo_utensils.svg';

class ComparisonResults extends Component {
  state = {
    protein: 0,
    fat: 0,
    carbohydrat: 0
  };

  static getDerivedStateFromProps(nextProps, prevState) {
    const formData = nextProps.formData;

    let p = (formData.protein + 1.5) / (formData.caloric / 10000);
    let f = (formData.fat + 1) / (formData.caloric / 10000);
    let c = (1000 - p * 3.5 - f * 8.5) / 3.5;
    c = c < 0 ? 0 : c;

    if (
      p !== prevState.protein ||
      f !== prevState.fat ||
      c !== prevState.carbohydrat
    ) {
      return {
        protein: p,
        fat: f,
        carbohydrat: c
      };
    }

    return null;
  }

  nomnomnowData(nnn) {
    const HBM = { protein: 80, fat: 62, carbohydrate: 55 };
    const PP = { protein: 78, fat: 63, carbohydrate: 56 };
    const CCW = { protein: 115, fat: 58, carbohydrate: 31 };
    const TTF = { protein: 96, fat: 46, carbohydrate: 78 };

    switch (nnn) {
      case 'Porkalicious Potluck':
        return PP;
      case 'Chicken Chow-Wow':
        return CCW;
      case 'Tasty Turkey Fare':
        return TTF;
      default:
        return HBM;
    }
  }

  render() {
    const { brand, nnn } = this.props.formData;
    const nnnFoodData = this.nomnomnowData(nnn);

    let otherBrand = (
      <div>
        <h3>{brand}</h3>
        <label>Protein</label>
        <h1>{this.state.protein}</h1>
        <label>Fat</label>
        <h1>{this.state.fat}</h1>
        <label>Carbohydrate</label>
        <h1>{this.state.carbohydrat}</h1>
      </div>
    );

    let nomnomnow = (
      <div>
        <img src={logo} alt="logo" />
        <h3>{nnn}</h3>
        <label>Protein *</label>
        <h1>{nnnFoodData.protein}</h1>
        <label>Fat</label>
        <h1>{nnnFoodData.fat}</h1>
        <label>Carbohydrate</label>
        <h1>{nnnFoodData.carbohydrate}</h1>
      </div>
    );

    return (
      <div>
        <div>{otherBrand}</div>
        <div>{nomnomnow}</div>
        <h5>
          Learn about the role of protein, fat, and carbs in a dog's diet.
        </h5>
        <h5>
          * All units above are given in grams per 1000 calories (g/kcal).
        </h5>
      </div>
    );
  }
}

export default ComparisonResults;
