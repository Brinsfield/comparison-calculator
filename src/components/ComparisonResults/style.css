.result-container {
  background-color: rgba(255, 255, 255, 0.8);
  padding: 20px;
  color: #818181;
}

.small-logo-icon {
  display: block;
  height: 40px;
  width: auto;
  margin-bottom: 10px;
}

.results {
  display: flex;
}

.results > div {
  padding: 20px 0 0 20px;
  width: 100%;
}
.results > div:first-child {
  border-right: 2px solid #e0e0e0;
  padding: 70px 20px 0 0;
  text-align: right;
}

.results > div:nth-child(2) {
  text-align: left;
}

h3.brand {
  font-weight: 300;
  line-height: 1.1em;
  height: 60px;
  margin: 0;
  font-size: 22px;
  color: #444444;
}

h1 {
  font-weight: 300;
  line-height: 1.1em;
  margin-bottom: 20px;
  font-size: 24px;
}

.color-orange {
  color: #f06324;
}
.color-teal {
  color: #23ac9b;
}

.color-yellow {
  color: #e2d16e;
}

a {
  color: #f06324;
  text-decoration: none;
  font-weight: 400;
}

.text-smallest {
  font-size: 13px;
}

.padding-bottom {
  padding-bottom: 20px;
}
