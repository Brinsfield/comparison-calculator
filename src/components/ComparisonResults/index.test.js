import React from 'react';
import { shallow } from 'enzyme'; // how to mount the component
import toJson from 'enzyme-to-json';
import ComparisonResults from '.';

describe('<ComparisonResults />', () => {
  let wrapper;
  it('renders', () => {
    let obj = {
      brand: 'Bonio',
      caloric: 3,
      fat: 2,
      nnn: 'Heartland Beef Mash',
      protein: 1
    };
    wrapper = shallow(<ComparisonResults formData={obj} />);
  });
  it('matches snapshot', () => {
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });
});
