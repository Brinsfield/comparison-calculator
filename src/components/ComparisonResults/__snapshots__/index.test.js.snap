// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComparisonResults /> matches snapshot 1`] = `
<div
  className="result-container "
>
  <div
    className="results padding-bottom"
  >
    <div>
      <aux>
        <h3
          className="brand"
        >
          Bonio
        </h3>
        <label
          className="label"
        >
          Protein
        </label>
        <h1>
          8333
        </h1>
        <label
          className="label"
        >
          Fat
        </label>
        <h1>
          10000
        </h1>
        <label
          className="label"
        >
          Carbohydrate
        </label>
        <h1>
          0
        </h1>
      </aux>
    </div>
    <div>
      <aux>
        <img
          alt="logo"
          className="small-logo-icon"
          src="logo_utensils.svg"
        />
        <h3
          className="brand"
        >
          Heartland Beef Mash
        </h3>
        <label
          className="label"
        >
          Protein *
        </label>
        <h1
          className="color-orange"
        >
          80
        </h1>
        <label
          className="label"
        >
          Fat
        </label>
        <h1
          className="color-teal"
        >
          62
        </h1>
        <label
          className="label"
        >
          Carbohydrate
        </label>
        <h1
          className="color-yellow"
        >
          55
        </h1>
      </aux>
    </div>
  </div>
  <div
    className="padding-bottom"
  >
    Learn about
     
    <a
      href="https://www.nomnomnow.com/learn/pet-expert/understanding-dog-food-nutrients"
    >
      the role of protein, fat, and carbs
    </a>
     
    in a dog's diet.
  </div>
  <div
    className="text-smallest"
  >
    * All units above are given in grams per 1000 calories (g/kcal).
  </div>
</div>
`;
